File: src/utils/format.ts
export function fmt_dur(ns: number): string {
  if (ns >= 1e9) return (ns / 1e9).toFixed(3) + ' s'
  if (ns >= 1e6) return (ns / 1e6).toFixed(1) + ' ms'
  return (ns / 1e3).toFixed(0) + ' \u00b5s'
}

export function fmt_pct(ns: number, total: number): string {
  if (total === 0) return '0.0%'
  return (ns / total * 100).toFixed(1) + '%'
}

File: android/app/src/main/java/com/swiperf/app/ui/util/Format.kt
package com.swiperf.app.ui.util

object Format {
    fun fmtDur(ns: Long): String {
        if (ns >= 1_000_000_000L) return "%.3f s".format(ns / 1e9)
        if (ns >= 1_000_000L) return "%.1f ms".format(ns / 1e6)
        return "${ns / 1000} \u00b5s"
    }

    fun fmtPct(ns: Long, total: Long): String {
        if (total == 0L) return "0.0%"
        return "%.1f%%".format(ns.toDouble() / total * 100)
    }
}
